images/virtualization-artifact/pkg/controller/watchers/vd_enqueuer.go

```diff
@@ -52,6 +52,39 @@ func (w VirtualDiskRequestEnqueuer) GetEnqueueFrom() client.Object {
 }
 
 func (w VirtualDiskRequestEnqueuer) EnqueueRequests(ctx context.Context, obj client.Object) (requests []reconcile.Request) {
+	// Enqueue CVI or VI specified by the object ref.
+	if w.enqueueFromKind == virtv2.VirtualDiskObjectRefKindVirtualImage {
+		vi, ok := obj.(*virtv2.VirtualImage)
```

Comment:
Open/Closed principle of SOLID is broken now for this method
It's better to dont
We can discuss how

Reply:
Fixed.